hydromt/drivers/dataframe/dataframe_driver.py

```diff
@@ -17,6 +21,7 @@ class DataFrameDriver(BaseDriver, ABC):
     def read(
         self,
         uri: str,
+        metadata: "SourceMetadata",
```

Comment:
@savente93 I can also make the metadata optional (would just have to move the Pydantic model to circumvent cyclic imports), what would you prefer?